File: DataProcessingSystem.java
import java.util.concurrent.*;
import java.util.*;
import java.io.*;

class Task {
    private final int id;

    public Task(int id) {
        this.id = id;
    }

    public int getId() {
        return id;
    }

    public String run() {
        try {
            Thread.sleep(100); // Simulate work delay
        } catch (InterruptedException e) {
            Thread.currentThread().interrupt();
            return "Task " + id + " interrupted during processing.";
        }
        return "Task " + id + " processed successfully.";
    }
}

class Worker implements Runnable {
    private final int workerId;
    private final BlockingQueue<Task> queue;
    private final List<String> results;

    public Worker(int workerId, BlockingQueue<Task> queue, List<String> results) {
        this.workerId = workerId;
        this.queue = queue;
        this.results = results;
    }

    @Override
    public void run() {
        System.out.println("Worker " + workerId + " started.");
        try {
            while (true) {
                Task task = queue.poll(1, TimeUnit.SECONDS);
                if (task == null) break;
                String output = task.run();
                synchronized (results) {
                    results.add(output);
                }
            }
        } catch (Exception ex) {
            System.err.println("Worker " + workerId + " encountered an error: " + ex.getMessage());
        }
        System.out.println("Worker " + workerId + " completed.");
    }
}

public class DataProcessingSystem {
    public static void main(String[] args) {
        BlockingQueue<Task> taskQueue = new LinkedBlockingQueue<>();
        List<String> resultList = Collections.synchronizedList(new ArrayList<>());

        for (int i = 1; i <= 10; i++) {
            taskQueue.add(new Task(i));
        }

        ExecutorService pool = Executors.newFixedThreadPool(3);

        for (int i = 1; i <= 3; i++) {
            pool.submit(new Worker(i, taskQueue, resultList));
        }

        pool.shutdown();
        try {
            pool.awaitTermination(10, TimeUnit.SECONDS);
        } catch (InterruptedException e) {
            System.err.println("Thread pool interrupted while waiting.");
        }

        try (BufferedWriter writer = new BufferedWriter(new FileWriter("results_java.txt"))) {
            for (String line : resultList) {
                writer.write(line);
                writer.newLine();
            }
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
        }

        System.out.println("Processing complete. Output written to results_java.txt");
    }
}

File: DataProcessingSystem.go
package main

import (
	"fmt"
	"log"
	"os"
	"sync"
	"time"
)

type Task struct {
	ID int
}

func (t Task) Run() string {
	time.Sleep(100 * time.Millisecond) // Simulated work
	return fmt.Sprintf("Task %d processed successfully.", t.ID)
}

func worker(id int, taskChan <-chan Task, results *[]string, mu *sync.Mutex, wg *sync.WaitGroup) {
	defer wg.Done()
	log.Printf("Worker %d started.\n", id)

	for task := range taskChan {
		result := task.Run()

		// Lock results
		mu.Lock()
		*results = append(*results, result)
		mu.Unlock()
	}

	log.Printf("Worker %d finished.\n", id)
}

func main() {
	tasks := make(chan Task, 10)
	var results []string
	var mu sync.Mutex
	var wg sync.WaitGroup

	for i := 1; i <= 10; i++ {
		tasks <- Task{ID: i}
	}
	close(tasks)

	for i := 1; i <= 3; i++ {
		wg.Add(1)
		go worker(i, tasks, &results, &mu, &wg)
	}

	wg.Wait()

	file, err := os.Create("results_go.txt")
	if err != nil {
		log.Fatalf("Could not create output file: %v", err)
	}
	defer file.Close()

	for _, line := range results {
		_, err := file.WriteString(line + "\n")
		if err != nil {
			log.Printf("Error writing result: %v", err)
		}
	}

	fmt.Println("All tasks completed. Output saved to results_go.txt")
}
